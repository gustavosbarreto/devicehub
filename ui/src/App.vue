<template>
<v-app>
    <v-navigation-drawer v-if="isLoggedIn" :clipped="false" fixed v-model="drawer" :mini-variant="false" enable-resize-watcher app>
        <v-container>
            <div class="text-center">
                <v-icon>mdi-console</v-icon>
                <h2 style="font-family: monospace">ShellHub</h2>
                <span class="overline">beta</span>
            </div>
        </v-container>
        <v-list>
            <v-list-item v-for="item in items" :key="item.title" :to="item.path" two-line>
                <v-list-item-action>
                    <v-icon v-text="item.icon"></v-icon>
                </v-list-item-action>

                <v-list-item-content>
                    <v-list-item-title v-text="item.title"></v-list-item-title>
                </v-list-item-content>
            </v-list-item>
        </v-list>
    </v-navigation-drawer>
    <v-app-bar app elevate-on-scroll color="primary" class="pl-3 pr-4" v-if="isLoggedIn">
        <v-menu transition="scale-transition" origin="top left">
            <template v-slot:activator="{ on }">
                <v-chip v-on="on">
                    <v-icon left>mdi-server</v-icon>
                    My Device Fleet
                    <v-icon right>mdi-chevron-down</v-icon>
                </v-chip>
            </template>
        </v-menu>
        <v-spacer></v-spacer>
        <v-chip>
            <v-icon>help</v-icon>
        </v-chip>
        <v-chip>
            <v-icon>notifications</v-icon>
        </v-chip>
        <v-menu transition="scale-transition" origin="top right">
            <template v-slot:activator="{ on }">
                <v-chip v-on="on">
                    <v-icon left>mdi-account</v-icon>
                    {{ $store.getters["auth/currentUser"] }}
                    <v-icon right>mdi-chevron-down</v-icon>
                </v-chip>
            </template>

            <v-card>
                <v-list-item three-line>
                    <v-list-item-content>
                        <v-list-item-title class="mb-1">Tenant ID</v-list-item-title>
                        <v-list-item-subtitle>
                            <v-chip>
                                <span>
                                    {{ tenant }}
                                </span>
                                <v-icon right v-clipboard="tenant" v-clipboard:success="() => { copySnack = true; }">mdi-content-copy</v-icon>
                            </v-chip>
                        </v-list-item-subtitle>
                    </v-list-item-content>
                </v-list-item>

                <v-card-actions>
                    <v-btn small text @click="logout()">
                        Logout
                    </v-btn>
                </v-card-actions>
            </v-card>
        </v-menu>
    </v-app-bar>
    <v-content>
        <v-container class="pa-8" fluid>
            <router-view></router-view>
        </v-container>
        <v-snackbar v-model="copySnack" :timeout=3000>Tenant ID copied to clipboard</v-snackbar>
    </v-content>
</v-app>
</template>

<script>
export default {
  name: "App",

  methods: {
    logout() {
      this.$store.dispatch("auth/logout").then(() => {
        this.$router.push("/login");
      });
    }
  },

  computed: {
    tenant() {
      return this.$store.getters["auth/tenant"];
    },

    isLoggedIn() {
      return this.$store.getters["auth/isLoggedIn"];
    }
  },

  data() {
    return {
      drawer: true,
      clipped: false,
      copySnack: false,
      items: [
        {
          icon: "dashboard",
          title: "Dashboard",
          path: "/"
        },
        {
          icon: "devices",
          title: "Devices",
          path: "/devices"
        },
        {
          icon: "history",
          title: "Sessions",
          path: "/sessions"
        }
      ]
    };
  }
};
</script>

<style>
.v-list-active {
  border-left: 4px solid var(--v-primary-base);
}

.text-shadow {
  text-shadow: #000 0 0 6px;
  color: transparent;
}
</style>
