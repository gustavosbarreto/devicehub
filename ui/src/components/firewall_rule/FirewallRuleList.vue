<template>
  <fragment>
    <v-card class="mt-2">
      <v-data-table
        :headers="headers"
        :items="getFirewallRules"
        item-key="uid"
        :sort-by="['started_at']"
        :sort-desc="[true]"
        :items-per-page="10"
        :footer-props="{'items-per-page-options': [10, 25, 50, 100]}"
        :server-items-length="getNumberFirewallRules"
        :options.sync="pagination"
        :disable-sort="true"
        data-test="firewallRuleList-dataTable"
      >
        <template #[`item.active`]="{ item }">
          <v-icon
            v-if="item.active"
            color="success"
          >
            check_circle
          </v-icon>
          <v-icon
            v-else
            bottom
          >
            check_circle
          </v-icon>
        </template>

        <template #[`item.priority`]="{ item }">
          {{ item.priority }}
        </template>

        <template #[`item.action`]="{ item }">
          {{ toUpperCase(item.action) }}
        </template>

        <template #[`item.source_ip`]="{ item }">
          {{ formatSourceIP(item.source_ip) }}
        </template>

        <template #[`item.username`]="{ item }">
          {{ formatUsername(item.username) }}
        </template>

        <template #[`item.filter`]="{ item }">
          <div
            v-if="filterKey(item.filter)=='hostname'"
          >
            {{ formatHostnameFilter(item.filter) }}
          </div>

          <div v-else-if="filterKey(item.filter)=='tags'">
            <v-tooltip
              v-for="(tag, index) in item.filter.tags"
              :key="index"
              bottom
              :disabled="!showTag(tag)"
            >
              <template #activator="{ on, attrs }">
                <v-chip
                  class="ml-1 mb-1"
                  small
                  outlined
                  v-bind="attrs"
                  v-on="on"
                >
                  {{ displayOnlyTenCharacters(tag) }}
                </v-chip>
              </template>

              <span v-if="showTag(tag)">
                {{ tag }}
              </span>
            </v-tooltip>
          </div>
        </template>

        <template #[`item.actions`]="{ item }">
          <v-menu
            :ref="'menu'+getFirewallRules.indexOf(item)"
            offset-y
          >
            <template #activator="{ on, attrs }">
              <v-chip
                color="transparent"
                v-on="on"
              >
                <v-icon
                  small
                  class="icons"
                  v-bind="attrs"
                  v-on="on"
                >
                  mdi-dots-horizontal
                </v-icon>
              </v-chip>
            </template>

            <v-card>
              <v-tooltip
                bottom
                :disabled="hasAuthorizationFormDialogEdit"
              >
                <template #activator="{ on, attrs }">
                  <div
                    v-bind="attrs"
                    v-on="on"
                  >
                    <v-list-item
                      :disabled="!hasAuthorizationFormDialogEdit"
                      @click.stop="showFirewallRuleEdit(getFirewallRules.indexOf(item))"
                    >
                      <FirewallRuleEdit
                        :firewall-rule="item"
                        :show.sync="firewallRuleEditShow[getFirewallRules.indexOf(item)]"
                        data-test="firewallRuleEdit-component"
                        @update="refresh"
                      />
                    </v-list-item>
                  </div>
                </template>

                <span>
                  You don't have this kind of authorization.
                </span>
              </v-tooltip>

              <v-tooltip
                bottom
                :disabled="hasAuthorizationFormDialogRemove"
              >
                <template #activator="{ on, attrs }">
                  <div
                    v-bind="attrs"
                    v-on="on"
                  >
                    <v-list-item
                      :disabled="!hasAuthorizationFormDialogRemove"
                      @click.stop="showFirewallRuleDelete(getFirewallRules.indexOf(item))"
                    >
                      <FirewallRuleDelete
                        :id="item.id"
                        :show.sync="firewallRuleDeleteShow[getFirewallRules.indexOf(item)]"
                        data-test="firewallRuleDelete-component"
                        @update="refresh"
                      />
                    </v-list-item>
                  </div>
                </template>

                <span>
                  You don't have this kind of authorization.
                </span>
              </v-tooltip>
            </v-card>
          </v-menu>
        </template>
      </v-data-table>
    </v-card>
  </fragment>
</template>

<script>

import FirewallRuleEdit from '@/components/firewall_rule/FirewallRuleFormDialogEdit';
import FirewallRuleDelete from '@/components/firewall_rule/FirewallRuleDelete';

import hasPermission from '@/components/filter/permission';

export default {
  name: 'FirewallRuleListComponent',

  filters: { hasPermission },

  components: {
    FirewallRuleDelete,
    FirewallRuleEdit,
  },

  data() {
    return {
      pagination: {},
      firewallRuleEditShow: [],
      firewallRuleDeleteShow: [],
      editAction: 'edit',
      removeAction: 'remove',
      headers: [
        {
          text: 'Active',
          value: 'active',
          align: 'center',
        },
        {
          text: 'Priority',
          value: 'priority',
          align: 'center',
        },
        {
          text: 'Action',
          value: 'action',
          align: 'center',
        },
        {
          text: 'Source IP',
          value: 'source_ip',
          align: 'center',
        },
        {
          text: 'Username',
          value: 'username',
          align: 'center',
        },
        {
          text: 'Filter',
          value: 'filter',
          align: 'center',
        },
        {
          text: 'Actions',
          value: 'actions',
          align: 'center',
        },
      ],

      showHelp: false,
    };
  },

  computed: {
    getFirewallRules() {
      return this.$store.getters['firewallrules/list'];
    },

    getNumberFirewallRules() {
      return this.$store.getters['firewallrules/getNumberFirewalls'];
    },

    hasAuthorizationFormDialogEdit() {
      const role = this.$store.getters['auth/role'];
      if (role !== '') {
        return hasPermission(
          this.$authorizer.role[role],
          this.$actions.firewall[this.editAction],
        );
      }

      return false;
    },

    hasAuthorizationFormDialogRemove() {
      const role = this.$store.getters['auth/role'];
      if (role !== '') {
        return hasPermission(
          this.$authorizer.role[role],
          this.$actions.firewall[this.removeAction],
        );
      }

      return false;
    },
  },

  watch: {
    pagination: {
      handler() {
        this.getFirewalls();
      },
      deep: true,
    },
  },

  methods: {
    refresh() {
      this.getFirewalls();
    },

    toUpperCase(str) {
      return str.charAt(0).toUpperCase().concat(str.slice(1));
    },

    displayOnlyTenCharacters(str) {
      if (str !== undefined) {
        if (str.length > 10) return `${str.substr(0, 10)}...`;
      }
      return str;
    },

    filterKey(filter) {
      return Reflect.ownKeys(filter)[0];
    },

    showTag(str) {
      if (str !== undefined) {
        if (str.length > 10) {
          return true;
        }
      }
      return false;
    },

    formatHostnameFilter(filter) {
      return filter.hostname === '.*' ? 'All devices' : filter.hostname;
    },

    formatSourceIP(ip) {
      return ip === '.*' ? 'Any IP' : ip;
    },

    formatUsername(username) {
      return username === '.*' ? 'All users' : username;
    },

    async getFirewalls() {
      if (!this.$store.getters['boxs/getStatus']) {
        const data = {
          perPage: this.pagination.itemsPerPage,
          page: this.pagination.page,
        };

        try {
          await this.$store.dispatch('firewallrules/fetch', data);

          this.setArrays();
        } catch (error) {
          if (error.response.status === 403) {
            this.$store.dispatch('snackbar/showSnackbarErrorAssociation');
          } else {
            this.$store.dispatch('snackbar/showSnackbarErrorLoading', this.$errors.snackbar.firewallRuleList);
          }
        }
      } else {
        this.setArrays();
        this.$store.dispatch('boxs/setStatus', false);
      }
    },

    showFirewallRuleEdit(index) {
      this.firewallRuleEditShow[index] = this.firewallRuleEditShow[index] === undefined
        ? true : !this.firewallRuleEditShow[index];
      this.$set(this.firewallRuleEditShow, index, this.firewallRuleEditShow[index]);

      this.closeMenu(index);
    },

    showFirewallRuleDelete(index) {
      this.firewallRuleDeleteShow[index] = this.firewallRuleDeleteShow[index] === undefined
        ? true : !this.firewallRuleDeleteShow[index];
      this.$set(this.firewallRuleDeleteShow, index, this.firewallRuleDeleteShow[index]);

      this.closeMenu(index);
    },

    setArrays() {
      const numberFirewallRules = this.getFirewallRules.length;

      if (numberFirewallRules > 0) {
        this.firewallRuleEditShow = new Array(numberFirewallRules).fill(false);
        this.firewallRuleDeleteShow = new Array(numberFirewallRules).fill(false);
      }
    },

    closeMenu(index) {
      this.$refs[`menu${index}`].isActive = false;
    },
  },
};
</script>
