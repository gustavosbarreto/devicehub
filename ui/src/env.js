export default {
  /* eslint-disable no-param-reassign */
  install(Vue) {
    const env = {
      isHosted: window.env.HOSTED === 'true',
    };

    Vue.env = env;
    Vue.prototype.$env = env;
  },
  /* eslint-enable no-param-reassign */
};
