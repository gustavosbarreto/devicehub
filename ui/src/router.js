import Vue from "vue";
import Router from "vue-router";
import Dashboard from "./views/Dashboard.vue";
import store from './store'

Vue.use(Router);

const router = new Router({
    mode: "history",
    base: process.env.BASE_URL,
    routes: [
        {
            path: "/",
            name: "dashboard",
            component: Dashboard
        },
        {
            path: "/devices",
            name: "devices",

            component: () =>
                import(/* webpackChunkName: "devices" */ "./views/Devices.vue")
        },
        {
            path: "/devices/:id",
            name: "featureDevice",
            component: () =>
                import("./views/FeatureDevice.vue") 
        },
        {
            path: "/sessions",
            name: "sessions",
            component: () =>
                import("./views/Sessions.vue")
        },
        {
            path: "/login",
            name: "login",
            component: () =>
                import("./views/Login.vue")
        }
    ]
});

router.beforeEach((to, from, next) => {
    if (to.path !== '/login') {
        if (store.getters['auth/isLoggedIn']) {
          return next()
        }

        return next(`/login?redirect=${to.path}`)
      } else {
        return next()
      }
})

export default router;