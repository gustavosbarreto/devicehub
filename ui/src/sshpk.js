const NodeRSA = require("node-rsa");
const sshpk = require("sshpk");

const createSignatureOfPrivateKey = (
  privateKeyData,
  username,
) => {
  const key = NodeRSA(privateKeyData);
  key.setOptions({ signingScheme: "pkcs1-sha1" });
  const signature = encodeURIComponent(key.sign(username, "base64"));
  return signature;
};

const createKeyFingerprint = (privateKeyData) => {
  const key = sshpk.parsePrivateKey(privateKeyData);
  const fingerprint = key.fingerprint("md5").toString("hex");
  return fingerprint;
};

const parsePrivateKey = (privateKey) => {
  const key = sshpk.parsePrivateKey(privateKey);
  return key;
};

const parseKey = (key) => {
  const parsedKey = sshpk.parseKey(key);
  return parsedKey;
};

const convertKeyToFingerprint = (privateKey) => {
  const fingerprint = sshpk.parsePrivateKey(privateKey).fingerprint("md5");
  return fingerprint;
};

const createSignerAndUpdate = (privateKey, username) => {
  const signer = privateKey.createSign("sha512");
  signer.update(username);
  return signer.sign().toString();
};

window.global.createSignatureOfPrivateKey = createSignatureOfPrivateKey;
window.global.createKeyFingerprint = createKeyFingerprint;
window.global.convertKeyToFingerprint = convertKeyToFingerprint;
window.global.createSignerAndUpdate = createSignerAndUpdate;
window.global.parsePrivateKey = parsePrivateKey;
window.global.parseKey = parseKey;
