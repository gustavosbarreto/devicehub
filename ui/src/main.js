import Vue from 'vue';

import Snackbar from '@/components/snackbar/Snackbar';

import App from './App';
import router from './router';
import store from './store';
import env from './env';
import './vee-validate';
import vuetify from './plugins/vuetify';

Vue.config.productionTip = false;

Vue.component('Snackbar', Snackbar);

Vue.use(require('vue-moment'));

Vue.use(env);

new Vue({
  vuetify,
  router,
  store,
  render: (h) => h(App),
}).$mount('#app');
