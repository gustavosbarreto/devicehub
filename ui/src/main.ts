import { createApp } from "vue";
import * as Sentry from "@sentry/vue";
import { BrowserTracing } from "@sentry/tracing";
import { envVariables } from "./envVariables";
import vuetify from "./plugins/vuetify";
import { key, store } from "./store";
import { router } from "./router";
import App from "./App.vue";

import { loadFonts } from "./plugins/webfontloader";

import SnackbarComponent from "./components/Snackbar/Snackbar.vue";
import { SnackbarPlugin } from "./plugins/snackbar";

const app = createApp(App);

Sentry.init({
  app,
  dsn: envVariables.sentryDsn || "",
  integrations: [
    new BrowserTracing({
      routingInstrumentation: Sentry.vueRouterInstrumentation(router),
    }),
  ],
  tracesSampleRate: 1.0,
  hooks: ["activate", "create", "destroy", "mount", "update"],
  timeout: 500,
  release: envVariables.version || "latest",
});
Sentry.setTag("project", "shellhub-ui");

loadFonts();

app.use(vuetify);
app.use(router);
app.use(store, key);
app.use(SnackbarPlugin);
app.component("SnackbarComponent", SnackbarComponent);
app.mount("#app");
