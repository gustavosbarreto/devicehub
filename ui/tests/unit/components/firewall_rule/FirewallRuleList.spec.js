import Vuex from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import FirewallRuleList from '@/components/firewall_rule/FirewallRuleList';
import { actions, authorizer } from '../../../../src/authorizer';

describe('FirewallRuleList', () => {
  const localVue = createLocalVue();
  const vuetify = new Vuetify();
  localVue.use(Vuex);

  let wrapper;

  const statusGlobal = true;
  const numberFirewallsGlobal = 2;

  const firewallsGlobal = [
    {
      id: '5f1996c8',
      tenant_id: 'xxxxxxxx',
      priority: 4,
      action: 'allow',
      active: true,
      source_ip: '00.00.00',
      username: 'shellhub',
      hostname: 'shellhub',
    },
    {
      id: '5f1996c8',
      tenant_id: 'xxxxxxxx',
      priority: 3,
      action: 'allow',
      active: false,
      source_ip: '00.00.00',
      username: 'shellhub',
      hostname: 'shellhub',
    },
  ];

  const headers = [
    {
      text: 'Active',
      value: 'active',
      align: 'center',
    },
    {
      text: 'Priority',
      value: 'priority',
      align: 'center',
    },
    {
      text: 'Action',
      value: 'action',
      align: 'center',
    },
    {
      text: 'Source IP',
      value: 'source_ip',
      align: 'center',
    },
    {
      text: 'Username',
      value: 'username',
      align: 'center',
    },
    {
      text: 'Hostname',
      value: 'hostname',
      align: 'center',
    },
    {
      text: 'Actions',
      value: 'actions',
      align: 'center',
    },
  ];

  const tests = [
    {
      description: 'List data when user has owner role',
      role: {
        type: 'owner',
        permission: true,
      },
      variables: {
        firewallsGlobal,
        numberFirewallsGlobal,
        statusGlobal,
      },
      data: {
        firewallRuleEditShow: [false, false],
        firewallRuleDeleteShow: [false, false],
        removeAction: 'remove',
        headers,
      },
      computed: {
        getFirewallRules: firewallsGlobal,
        getNumberFirewallRules: numberFirewallsGlobal,
        hasAuthorizationFormDialogRemove: true,
      },
    },
    {
      description: 'List data when user has operator role',
      role: {
        type: 'operator',
        permission: false,
      },
      variables: {
        firewallsGlobal,
        numberFirewallsGlobal,
        statusGlobal,
      },
      data: {
        firewallRuleEditShow: [false, false],
        firewallRuleDeleteShow: [false, false],
        removeAction: 'remove',
        headers,
      },
      computed: {
        getFirewallRules: firewallsGlobal,
        getNumberFirewallRules: numberFirewallsGlobal,
        hasAuthorizationFormDialogRemove: false,
      },
    },
  ];

  const storeVuex = (firewalls, numberFirewalls, status, currentrole) => new Vuex.Store({
    namespaced: true,
    state: {
      firewalls,
      numberFirewalls,
      status,
      currentrole,
    },
    getters: {
      'firewallrules/list': (state) => state.firewalls,
      'firewallrules/getNumberFirewalls': (state) => state.numberFirewalls,
      'boxs/getStatus': (state) => state.status,
      'auth/role': (state) => state.currentrole,
    },
    actions: {
      'firewallrules/fetch': () => {},
      'boxs/setStatus': () => {},
      'snackbar/showSnackbarErrorAssociation': () => {},
      'snackbar/showSnackbarErrorLoading': () => {},
    },
  });

  tests.forEach((test) => {
    describe(`${test.description}`, () => {
      beforeEach(() => {
        wrapper = mount(FirewallRuleList, {
          store: storeVuex(
            test.variables.firewallsGlobal,
            test.variables.numberFirewallsGlobal,
            test.variables.statusGlobal,
            test.role.type,
          ),
          localVue,
          stubs: ['fragment'],
          vuetify,
          mocks: {
            $authorizer: authorizer,
            $actions: actions,
          },
        });
      });

      ///////
      // Component Rendering
      //////

      it('Is a Vue instance', () => {
        expect(wrapper).toBeTruthy();
      });
      it('Renders the component', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });

      ///////
      // Data and Props checking
      //////

      it('Compare data with default value', () => {
        Object.keys(test.data).forEach((item) => {
          expect(wrapper.vm[item]).toEqual(test.data[item]);
        });
      });
      it('Process data in the computed', () => {
        Object.keys(test.computed).forEach((item) => {
          expect(wrapper.vm[item]).toEqual(test.computed[item]);
        });
      });

      //////
      // HTML validation
      //////

      it('Renders the template with data', () => {
        const dt = wrapper.find('[data-test="firewallRuleList-dataTable"]');
        const dataTableProps = dt.vm.$options.propsData;

        expect(dataTableProps.items).toHaveLength(test.variables.numberFirewallsGlobal);
      });
    });
  });
});
