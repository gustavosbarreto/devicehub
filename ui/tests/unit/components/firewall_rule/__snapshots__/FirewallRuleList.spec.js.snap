// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirewallRuleList List data when user has operator role Renders the component 1`] = `
<fragment-stub>
  <div class="mt-2 v-card v-sheet theme--light">
    <div class="v-data-table v-data-table--has-bottom theme--light" data-test="firewallRuleList-dataTable">
      <div class="v-data-table__wrapper">
        <table>
          <colgroup>
            <col class="">
            <col class="">
            <col class="">
            <col class="">
            <col class="">
            <col class="">
            <col class="">
          </colgroup>
          <thead class="v-data-table-header">
            <tr>
              <th role="columnheader" scope="col" aria-label="Active" class="text-center"><span>Active</span></th>
              <th role="columnheader" scope="col" aria-label="Priority" class="text-center"><span>Priority</span></th>
              <th role="columnheader" scope="col" aria-label="Action" class="text-center"><span>Action</span></th>
              <th role="columnheader" scope="col" aria-label="Source IP" class="text-center"><span>Source IP</span></th>
              <th role="columnheader" scope="col" aria-label="Username" class="text-center"><span>Username</span></th>
              <th role="columnheader" scope="col" aria-label="Filter" class="text-center"><span>Filter</span></th>
              <th role="columnheader" scope="col" aria-label="Actions" class="text-center"><span>Actions</span></th>
            </tr>
          </thead>
          <tbody>
            <tr class="">
              <td class="text-center"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light success--text">check_circle</i></td>
              <td class="text-center">
                4
              </td>
              <td class="text-center">
                Allow
              </td>
              <td class="text-center">
                00.00.00
              </td>
              <td class="text-center">
                shellhub
              </td>
              <td class="text-center">
                <div>
                  shellhub
                </div>
              </td>
              <td class="text-center"><span class="v-chip v-chip--clickable theme--light v-size--default transparent"><span class="v-chip__content"><button type="button" role="button" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate icons v-icon--link mdi mdi-dots-horizontal theme--light" style="font-size: 16px;"></button></span></span>
                <div class="v-menu">
                  <!---->
                </div>
              </td>
            </tr>
            <tr class="">
              <td class="text-center"><i aria-hidden="true" bottom="" class="v-icon notranslate material-icons theme--light">check_circle</i></td>
              <td class="text-center">
                3
              </td>
              <td class="text-center">
                Allow
              </td>
              <td class="text-center">
                00.00.00
              </td>
              <td class="text-center">
                shellhub
              </td>
              <td class="text-center">
                <div><span class="ml-1 mb-1 v-chip v-chip--no-color v-chip--outlined theme--light v-size--small" aria-haspopup="true" aria-expanded="false"><span class="v-chip__content">
                tag1
              </span></span><span class="v-tooltip v-tooltip--bottom"><!----></span><span class="v-tooltip v-tooltip--bottom"><!----></span><span class="ml-1 mb-1 v-chip v-chip--no-color v-chip--outlined theme--light v-size--small" aria-haspopup="true" aria-expanded="false"><span class="v-chip__content">
                tag2
              </span></span></div>
              </td>
              <td class="text-center"><span class="v-chip v-chip--clickable theme--light v-size--default transparent"><span class="v-chip__content"><button type="button" role="button" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate icons v-icon--link mdi mdi-dots-horizontal theme--light" style="font-size: 16px;"></button></span></span>
                <div class="v-menu">
                  <!---->
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div class="v-data-footer">
        <div class="v-data-footer__select">Rows per page:<div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select">
            <div class="v-input__control">
              <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-205" class="v-input__slot">
                <div class="v-select__slot">
                  <div class="v-select__selections">
                    <div class="v-select__selection v-select__selection--comma">10</div><input aria-label="Rows per page:" id="input-205" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
                  </div>
                  <div class="v-input__append-inner">
                    <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                  </div><input type="hidden" value="10">
                </div>
                <div class="v-menu">
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="v-data-footer__pagination">1-2 of 2</div>
        <div class="v-data-footer__icons-before"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Previous page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-left theme--light"></i></span></button></div>
        <div class="v-data-footer__icons-after"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Next page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-right theme--light"></i></span></button></div>
      </div>
    </div>
  </div>
</fragment-stub>
`;

exports[`FirewallRuleList List data when user has owner role Renders the component 1`] = `
<fragment-stub>
  <div class="mt-2 v-card v-sheet theme--light">
    <div class="v-data-table v-data-table--has-bottom theme--light" data-test="firewallRuleList-dataTable">
      <div class="v-data-table__wrapper">
        <table>
          <colgroup>
            <col class="">
            <col class="">
            <col class="">
            <col class="">
            <col class="">
            <col class="">
            <col class="">
          </colgroup>
          <thead class="v-data-table-header">
            <tr>
              <th role="columnheader" scope="col" aria-label="Active" class="text-center"><span>Active</span></th>
              <th role="columnheader" scope="col" aria-label="Priority" class="text-center"><span>Priority</span></th>
              <th role="columnheader" scope="col" aria-label="Action" class="text-center"><span>Action</span></th>
              <th role="columnheader" scope="col" aria-label="Source IP" class="text-center"><span>Source IP</span></th>
              <th role="columnheader" scope="col" aria-label="Username" class="text-center"><span>Username</span></th>
              <th role="columnheader" scope="col" aria-label="Filter" class="text-center"><span>Filter</span></th>
              <th role="columnheader" scope="col" aria-label="Actions" class="text-center"><span>Actions</span></th>
            </tr>
          </thead>
          <tbody>
            <tr class="">
              <td class="text-center"><i aria-hidden="true" class="v-icon notranslate material-icons theme--light success--text">check_circle</i></td>
              <td class="text-center">
                4
              </td>
              <td class="text-center">
                Allow
              </td>
              <td class="text-center">
                00.00.00
              </td>
              <td class="text-center">
                shellhub
              </td>
              <td class="text-center">
                <div>
                  shellhub
                </div>
              </td>
              <td class="text-center"><span class="v-chip v-chip--clickable theme--light v-size--default transparent"><span class="v-chip__content"><button type="button" role="button" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate icons v-icon--link mdi mdi-dots-horizontal theme--light" style="font-size: 16px;"></button></span></span>
                <div class="v-menu">
                  <!---->
                </div>
              </td>
            </tr>
            <tr class="">
              <td class="text-center"><i aria-hidden="true" bottom="" class="v-icon notranslate material-icons theme--light">check_circle</i></td>
              <td class="text-center">
                3
              </td>
              <td class="text-center">
                Allow
              </td>
              <td class="text-center">
                00.00.00
              </td>
              <td class="text-center">
                shellhub
              </td>
              <td class="text-center">
                <div><span class="ml-1 mb-1 v-chip v-chip--no-color v-chip--outlined theme--light v-size--small" aria-haspopup="true" aria-expanded="false"><span class="v-chip__content">
                tag1
              </span></span><span class="v-tooltip v-tooltip--bottom"><!----></span><span class="v-tooltip v-tooltip--bottom"><!----></span><span class="ml-1 mb-1 v-chip v-chip--no-color v-chip--outlined theme--light v-size--small" aria-haspopup="true" aria-expanded="false"><span class="v-chip__content">
                tag2
              </span></span></div>
              </td>
              <td class="text-center"><span class="v-chip v-chip--clickable theme--light v-size--default transparent"><span class="v-chip__content"><button type="button" role="button" aria-haspopup="true" aria-expanded="false" class="v-icon notranslate icons v-icon--link mdi mdi-dots-horizontal theme--light" style="font-size: 16px;"></button></span></span>
                <div class="v-menu">
                  <!---->
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div class="v-data-footer">
        <div class="v-data-footer__select">Rows per page:<div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select">
            <div class="v-input__control">
              <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-55" class="v-input__slot">
                <div class="v-select__slot">
                  <div class="v-select__selections">
                    <div class="v-select__selection v-select__selection--comma">10</div><input aria-label="Rows per page:" id="input-55" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
                  </div>
                  <div class="v-input__append-inner">
                    <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                  </div><input type="hidden" value="10">
                </div>
                <div class="v-menu">
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="v-data-footer__pagination">1-2 of 2</div>
        <div class="v-data-footer__icons-before"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Previous page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-left theme--light"></i></span></button></div>
        <div class="v-data-footer__icons-after"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Next page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-right theme--light"></i></span></button></div>
      </div>
    </div>
  </div>
</fragment-stub>
`;
